spec/models/annotation_text_spec.rb

```diff
@@ -24,4 +24,135 @@
       end
     end
   end
+
+  let(:assignment) { create(:assignment_with_deductive_annotations) }
+  let(:annotation_category_with_criterion) do
+    assignment.annotation_categories.where.not(flexible_criterion_id: nil).first
+  end
+  let(:deductive_text) do
+    annotation_category_with_criterion.annotation_texts.where.not(deduction: nil).first
+  end
+  let(:category_without_criteria) { create(:annotation_category, assignment: assignment) }
+  let(:text_without_deduction) { create(:annotation_text, annotation_category: category_without_criteria) }
+
```

Comment:
Add a variable for `:flexible_criterion` and use it below (so you don't have to keep doing `annotation_category_with_criterion.flexible_criterion`).